clients/vault/src/system.rs

```diff
@@ -558,3 +520,42 @@ pub(crate) async fn is_vault_registered(
 		Err(err) => Err(err.into()),
 	}
 }
+
+/// Returns SCPMessageHandler which contains the thread to connect/listen to the Stellar
+/// Node. See the oracle.rs example
+async fn inner_create_handler(
+	spacewalk_parachain: &SpacewalkParachain,
+	stellar_vault_secret_key: &str,
+) -> Result<ScpMessageHandler, Error> {
+	prepare_directories().map_err(|e| {
+		tracing::error!("Failed to create the SCPMessageHandler: {:?}", e);
+		Error::StellarSdkError
+	})?;
+
+	let is_public_net = spacewalk_parachain.is_public_network().await.map_err(Error::from)?;
+
+	let tier1_node_ip =
+		if is_public_net { TIER_1_VALIDATOR_IP_PUBLIC } else { TIER_1_VALIDATOR_IP_TESTNET };
+
+	let network: &Network = if is_public_net { &PUBLIC_NETWORK } else { &TEST_NETWORK };
+
+	tracing::info!(
+		"Connecting to {:?} through {:?}",
+		std::str::from_utf8(network.get_passphrase().as_slice()).unwrap(),
+		tier1_node_ip
+	);
+
+	let secret = SecretKey::from_encoding(stellar_vault_secret_key).unwrap();
+	let public_key_binary = hex::encode(secret.get_public().as_binary());
+	tracing::info!("public key binary {:?}", public_key_binary);
+
+	let node_info = NodeInfo::new(19, 21, 19, "v19.1.0".to_string(), network);
+	let cfg = ConnConfig::new(tier1_node_ip, 11625, secret, 0, true, true, false);
+
+	// todo: add vault addresses filter
+	let addresses = vec![];
```

Comment:
Can't we actually add the public key of our stellar account ie `secret.get_public()` here? Or what are these addresses for again?